lib/livebook/session/data.ex

```diff
@@ -1521,6 +1538,26 @@ defmodule Livebook.Session.Data do
     set!(data_actions, secrets: secrets)
   end
 
+  defp delete_secret({data, _} = data_actions, secret_name) do
+    idx = Enum.find_index(data.secrets, &(&1.name == secret_name))
+    secrets = List.delete_at(data.secrets, idx) |> Enum.sort()
```

Comment:
The secrets should be sorted already, so when we remove a secret, they are still sorted :D

```suggestion
    secrets = Enum.reject(data.secrets, &(&1.name == secret_name))
```